src/factory/deployModuleFactory.ts

```diff
@@ -21,10 +23,23 @@ export const deployModuleFactory = async (
 ): Promise<string> => {
   const singletonFactory = await getSingletonFactory(hre);
   console.log("    Singleton Factory:     ", singletonFactory.address);
-  const Factory = await hre.ethers.getContractFactory("ModuleProxyFactory");
+
+  try {
+    const Factory = await hre.ethers.getContractFactory("ModuleProxyFactory");
+    if (Factory.bytecode !== FactoryInitCode) {
+      console.warn(
+        "WARNING: The ModuleProxyFactory init code from (src/factory/contracts.ts) " +
+          "MasterCopyInitData[KnownContracts.FACTORY].initCode does not match the init " +
+          "code of the contract code at contracts/factory/ModuleProxyFactory.sol. " +
+          "You are most likely the MasterCopyInitData[KnownContracts.FACTORY].initCode is outdated."
+      );
+    }
+  } catch (e) {
+    // This is expected when the zodiac package is imported as a package.
```

Comment:
don't quite get this catch?

Reply:
When the zodiac package is installed as an NPM package (where the `ModuleProxyFactory.sol` contract is not available in the specified contract path), `const Factory = await hre.ethers.getContractFactory("ModuleProxyFactory");` will fail. 
Since the `ModuleProxyFactory` contract will not be available. So this check can only be performed while the function is used from within the zodiac project. When this is used as a package it will always go to the catch, but still should not fail.

Reply:
I think this is more like, there won't be an artifact cached on disk (no local hardhat compile will exist), and the getContractFactory method will fail?

Reply:
In any case, I think this will most likely be called be called by the singleton-deployment hardhat task, which I can only imagine being called by someone operating directly in the zodiac package? Or do you envision another use case?

So in any case, I think check is useful, it is a reminder to a maintainer to update the config objects, before deploying the next factory. I guess I would simply remove the try-catch

Reply:
> In any case, I think this will most likely be called by the singleton-deployment hardhat task, which I can only imagine being called by someone operating directly in the zodiac package? Or do you envision another use case?

It is useful when testing zodiac modules on locale test networks, for instance, when developing Zodiac Modules. I will be used during locale testing in the zodiac starter kit without the developers having to manually deploy the singleton factory and the module factory themselves.